src/commonMain/kotlin/exchange.dydx.abacus/validator/trade/TradeBracketOrdersValidator.kt

```diff
@@ -366,6 +707,39 @@ internal class TradeBracketOrdersValidator(
     }
 
     private fun validateStopLossReduceOnly(
+        trade: InternalTradeInputState,
+        position: InternalPerpetualPosition,
+    ): ValidationError? {
+        val reduceOnly = trade.brackets?.stopLoss?.reduceOnly ?: false
+        val sizePostOrder = position.calculated[CalculationPeriod.post]?.size ?: Numeric.double.ZERO
+        return if (reduceOnly) {
+            when (trade.side) {
+                OrderSide.Sell -> {
+                    if (sizePostOrder > Numeric.double.ZERO) {
+                        reduceOnlyError(
+                            field = listOf("brackets.stopLoss.triggerPrice", "brackets.stopLoss.reduceOnly"),
+                        )
+                    } else {
+                        null
+                    }
+                }
+                OrderSide.Buy -> {
+                    if (sizePostOrder < Numeric.double.ZERO) {
```

Comment:
For validating reduceOnly shouldn't we also compare sizePostOrder with the currentPositionSize?

Is there a separate conditional that handles that and this is only for the case in which they increase a position that is crossing?

Reply:
Yes, it's in TradePositionStateValidator